### PR TITLE
Increase visibility on code.gouv.fr

### PR DESCRIPTION
**Pull request by loicguilmard**
I filled the publiccode.yml describe the project respecting the metadata expected format.

### CODE REVIEW

**Comment by seb5g** [comment]
What's the purpose of this? It seems redundant with the readme and the pyproject.toml! Also it contains a fixed version number we will need to change every time. Is there a way to fill it dynamically from other source?

**Comment by loicguilmard** [comment]
I don't really know how influencial is the website but it would push PyMoDAQ a closer to the top score ratings. For the version, i don't think there is an included (in publiccode.yml) way for autoversion. I think i'll have to do it with github actions. I'll update this PR about it.

**Comment by seb5g** [comment]
ok but then could you explain what it is for this code.gouv.fr?

**Comment by loicguilmard** [comment]
code.gouv.fr is the result of the opensource politics in the french centralized administration to increase visibility and distribution of public opensource projects.
code.gouv.fr with the collaboration of DINUM implemented some quality measurements of opensource projects based on a few files :
publiccode.yml for all the metadata describing the project, where to find it, who to contact.
and a few markdown to get the "awesome" status and be on the frontpage of code.gouv.fr :
ROADMAP.md
AUTHORS.md
CONTRIBUTING.md
CHANGELOG.md
CODE_OF_CONDUCT.md
GOVERNANCE.md

I don't know how big or important it is. I just think it's gratifiying to see PyMoDAQ right next to miasm (CEA) or onyxia (INSEE)...

**Comment by seb5g** [comment]
ok then! But it could be nice if the markdown could be generated automatically to reflect changes if need be!

**Comment by loicguilmard** [comment]
could AUTHORS and CONTRIBUTING just markdown files that you include in docs/dev../contributing.rst ... ?
All the other files are statics.
For the publiccode.yml, i need to PR a small script building the script for git tag version.

**Comment by seb5g** [comment]
> could AUTHORS and CONTRIBUTING just markdown files that you include in docs/dev../contributing.rst ... ? All the other files are statics. For the publiccode.yml, i need to PR a small script building the script for git tag version.

yes could do, but now pyproject.toml also include a author field and a maintainer one and maybe a contributer one... but I agree it is not a big deal to do so. could you PR this also?

File: AUTHORS.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+Here is a list of the main contributors:
+
+## Main modules
+***************
+
+Functionalities
+---------------
+
+* Sébastien Weber, Research Engineer at CEMES/CNRS
```
Comment (seb5g) [inline AUTHORS.md:9]: should we update those lists from all the PR that have been made lately? Can you get that from git?

File: Contributing.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+* Talk about PyMoDAQ to your colleagues and with us
+  - [Forum](https://github.com/orgs/PyMoDAQ/discussions)
+  - [Mail](https://listes.services.cnrs.fr/wws/info/pymodaq)
+  - [Discord](https://discord.gg/mSrEwARwgm)
+* Use PyMoDAQ and report [issues](https://github.com/PyMoDAQ/PyMoDAQ/issues)
+* Cite PyMoDAQ in your papers [reference](CITATION.cff)
+* Easily add your instruments in plugins using the [template](https://github.com/PyMoDAQ/pymodaq_plugins_template)
+* Work on new features, on solving bugs or issues [PyMoDAQ Forge](https://github.com/PyMoDAQ)
+
+For a up-to-date detailled wallthrough, please visit the documentation
```
Comment (seb5g) [inline Contributing.md:15]: walkthrough?

File: docs/src/developer_folder/contributing.rst [hunk 3]
```diff
@@ -115,53 +115,12 @@ Math functions in ROI
...
-* Sébastien Weber, Research Engineer at CEMES/CNRS
-* Matthieu Cabos helped with this documentation
-* David Bresteau wrote the documentation of the PID extension and the tutorial: :ref:`plugin_development`
-
-Testing
-*******
-* Sébastien Weber, Research Engineer at CEMES/CNRS
-* Pierre Jannot wrote tests with a total of 5000 lines of code tested during his internship at CEMES in 2021
-
+.. include:: ../../../AUTHORS.md
```
Comment (seb5g) [inline docs/src/developer_folder/contributing.rst:123]: is it working to include markdown in rst?

**Comment by loicguilmard** [comment]
I used this AUTHORS update to reference myself in the cleaning/documentation contributors. 
I hope that's ok :D

I tried to write down a minimalist Code of Conduct. If we decide to adopt a specific one, we should all talk about it ?
I wrote a static Contributing just referencing all the tool we use, so we shouldn't have to change it often.
I didn't feel like writing down about Governance or Roadmap. It's above my League :D

About the version in the publiccode.yml, there is already an issue raised about automatic versioning [Publiccode.yml Issue 196](https://github.com/publiccodeyml/publiccode.yml/issues/196)

**Review by seb5g — APPROVED** [review]

**Comment by seb5g** [comment]
can you do the same PR on v5 (updating the few line that should be updated)? Also about the version keep a track on the issue you mentioned!

File: Contributing.md (hunk 2 above)
Comment (loicguilmard) [reply Contributing.md:15]: oups yes typo

File: docs/src/developer_folder/contributing.rst (hunk 3 above)
Comment (loicguilmard) [reply docs/src/developer_folder/contributing.rst:123]: i tested the include. It's working.
I just needed to add ## in front of some title that appears in the rst

**Pull request merged**